### PR TITLE
Feature/initial implementation

### PR DESCRIPTION
**Pull request by sfrycertinia**
Redis pub/sub transport implementation.

### CODE REVIEW

File: doc/index.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    
+
+
+
+    
+
+
+    <h3> </h3>
+
```
Comment (dfruddffdc) [inline doc/index.html:30]: Why so empty?

File: doc/index.js.html [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    
+
+
+
+    
+    <section>
+        <article>
+            <pre class="prettyprint source linenums"><code>/**
+ * Copyright (c) 2017, FinancialForce.com, inc
```
Comment (dfruddffdc) [inline doc/index.js.html:30]: Do we need so much copyright stuff in the documentation examples?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@financialforcedev/orizuru-transport-redis",
+  "version": "3.0.3",
```
Comment (dfruddffdc) [inline package.json:3]: Version number seems high for a new package. Perhaps it should be 0.something?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    "heroku",
+    "worker",
+    "dyno",
+    "orizuru"
+  ],
+  "author": "financialforce",
+  "license": "BSD-3-Clause",
+  "dependencies": {
+    "lodash": "4.17.4",
+    "redis": "^2.8.0"
```
Comment (dfruddffdc) [inline package.json]: Carrot.

File: src/lib/index/subscribe.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	ERROR_EVENT = 'error_event',
+
+	emitter = new EventEmitter(),
+
+	wrapper = (handler) => (message) => {
+		// Invoke the handler with the message
+		try {
+			const handlerResult = handler(message);
+			return Promise.resolve(handlerResult).catch(err => {
+				emitter.emit(ERROR_EVENT, err.message);
```
Comment (dfruddffdc) [inline src/lib/index/subscribe.js]: Not keen on the duplicate catch behaviour. Couldn't you do..

    return Promise.resolve(message)
        .then(handler)
        .catch(err => {
            emitter.emit(ERROR_EVENT, err.message);
            return Promise.reject(err);
        });

File: src/spec/index/subscribe.spec.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+				// when - then
+				return expect(Subscriber.subscribe({})).to.be.rejectedWith('Fail');
+
+			});
+
+			it('if handler function throws an error', () => {
+
+				// given
+				const
```
Comment (dfruddffdc) [inline src/spec/index/subscribe.spec.js]: Would prefer single line const declarations when there's only 1, but not sure where this conversation ended up..

**Review by dfruddffdc — DISMISSED** [review]
All minor changes, happy enough to approve.

File: package.json (hunk 3 above)
Comment (sfrycertinia) [reply package.json:3]: Yes!

File: doc/index.html (hunk 1 above)
Comment (sfrycertinia) [reply doc/index.html:30]: Not sure how to get anything in, this seems the same in the other repos...

File: doc/index.js.html (hunk 2 above)
Comment (sfrycertinia) [reply doc/index.js.html:30]: This is just coming from the source files.

File: src/lib/index/subscribe.js (hunk 5 above)
Comment (sfrycertinia) [reply src/lib/index/subscribe.js]: Looks better, agreed.

**Pull request merged**